client/src/pages/PollPage.js

```diff
@@ -34,99 +34,50 @@ class PollPage extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            drawerIsOpen: true,
             pollDialogIsOpen: false,
-            hasVoted: false,
-            voted: [
-                {
-                    userId: 1,
-                    name: "Alison Brown",
-                    avatar: "https://randomuser.me/api/portraits/women/2.jpg",
-                    option: 1
-                },
-                {
-                    userId: 2,
-                    name: "Caroline Chapman",
-                    avatar: "https://randomuser.me/api/portraits/women/3.jpg",
-                    option: 0
-                },
-                {
-                    userId: 3,
-                    name: "Dorothy Ferguson",
-                    avatar: "https://randomuser.me/api/portraits/women/4.jpg",
-                    option: 1
-                },
-                {
-                    userId: 4,
-                    name: "Jack Murray",
-                    avatar: "https://randomuser.me/api/portraits/men/4.jpg",
-                    option: 1
-                }
-            ] // fake data for poll results
+            results: []
         };
     }
 
     componentDidMount() {
+        // TODO: show results array updaing in real time
         // get voting data on the poll from the database:
         const { _id } = this.props.location.state.poll;
-
         axios
             .get("/api/poll/results", {
                 params: {
-                    id: _id
+                    pollId: _id
                 }
             })
             .then(response => {
-                if (response.data.status === 200) {
-                    const hasVoted = response.data.find(
-                        voter => voter.userId === this.props.user._id
-                    );
-                    this.setState({ voted: response.data, hasVoted });
+                if ((response.status = 200)) {
+                    this.setState({ results: response.data.results });
```

Comment:
This won't work as you are trying to change the array, in my test I changed it to `this.state.results.push(response.data.results)`  but all that is add multiple results to this array so it ended up looking like [array1[arra1]] because there are multiple renders that force this item to update twice.  In the end I ended up commenting out this line altogether to get the card to display.

Reply:
It looks like this should be: `this.setState({ results: response.data.results });` ?

The backend sends an object with keys status and data :
` res.send({ status: 200, data: results }`

Reply:
shouldn't be*

Reply:
`response.data.data` then

Reply:
Oh, but on Phi's PR branch, the server response has been changed